### PR TITLE
Supporting frontends not having GET_PREFERRED_HW_RENDER.

### PR DESCRIPTION
**Pull request by Rinnegatamante**

### CODE REVIEW

File: Source/Core/DolphinLibretro/Video.cpp [hunk 1]
```diff
@@ -219,27 +219,54 @@ void Init()
...
-        if (Options::renderer == "Hardware")
-        	SConfig::GetInstance().m_strVideoBackend = "OGL";
-        else
-        	SConfig::GetInstance().m_strVideoBackend = "Software Renderer";
-        return;
+	
+	// Skipping the negotiation if frontend does not support that API call yet
+    if (!environ_cb(RETRO_ENVIRONMENT_GET_PREFERRED_HW_RENDER, &preferred)) preferred == 0xFFFFFFFF;
+	
+    if (preferred == RETRO_HW_CONTEXT_OPENGL || preferred == RETRO_HW_CONTEXT_OPENGL_CORE || preferreed == 0xFFFFFFFF) {
```
Comment (flyinghead) [inline Source/Core/DolphinLibretro/Video.cpp]: Looks like a typo here: `preferreed`

**Review by flyinghead — CHANGES_REQUESTED** [review]

File: Source/Core/DolphinLibretro/Video.cpp (hunk 1 above)
Comment (Rinnegatamante) [reply Source/Core/DolphinLibretro/Video.cpp]: Thanks, rebased the commit

File: Source/Core/DolphinLibretro/Video.cpp [hunk 2]
```diff
@@ -219,27 +219,54 @@ void Init()
...
+                  SConfig::GetInstance().m_strVideoBackend = "Software Renderer";
+               return;
+            }
+         }
       }
     }
 	
 #ifdef _WIN32
-    if (preferred == RETRO_HW_CONTEXT_DIRECT3D) {
+    if (preferred == RETRO_HW_CONTEXT_DIRECT3D || preferreed == 0xFFFFFFFF) {
```
Comment (Tatsuya79) [inline Source/Core/DolphinLibretro/Video.cpp]: typo here too

File: Source/Core/DolphinLibretro/Video.cpp [hunk 3]
```diff
@@ -252,7 +279,7 @@ void Init()
     }
 #endif
 #ifndef __APPLE__
-    if (preferred == RETRO_HW_CONTEXT_VULKAN) {
+    if (preferred == RETRO_HW_CONTEXT_VULKAN || preferreed == 0xFFFFFFFF) {
```
Comment (Tatsuya79) [inline Source/Core/DolphinLibretro/Video.cpp]: and here

**Review by flyinghead — APPROVED** [review]

**Pull request merged**